Lib/test/test_httplib.py

```diff
@@ -2390,6 +2390,29 @@ def test_tunnel_debuglog(self):
         lines = output.getvalue().splitlines()
         self.assertIn('header: {}'.format(expected_header), lines)
 
+    def test_tunnel_leak(self):
+        sock = None
+
+        def _create_connection(address, timeout=None, source_address=None):
+            nonlocal sock
+            sock = FakeSocket(
+                'HTTP/1.1 404 NOT FOUND\r\n\r\n',
+                host=address[0],
+                port=address[1],
+            )
+            return sock
+
+        self.conn._create_connection = _create_connection
+        self.conn.set_tunnel('destination.com')
```

Comment:
as a failsafe, use a domain name we control in the test so that if something goes wrong in code (no connection or DNS resolution of this host is _supposed_ to happen in this test) we're not sending broken test traffic elsewhere.

Reply:
I copied the other tests, should I update them all?

Reply:
oh I didn't notice it was in the others.  nevermind then.  they can all be updated at once if it's ever an issue.